core-build/gulp-core-build/src/logging.ts

```diff
@@ -471,17 +471,20 @@ const colorCodeRegex: RegExp = /\x1B[[(?);]{0,2}(;?\d)*./g;
 
 /**
  * Adds a suppression for an error or warning
- * @param str - the error or warning as a string
+ * @param str - the error or warning as a string or Regular Expression
  * @public
  */
-export function addSuppression(str: string): void {
+export function addSuppression(str: string | RegExp): void {
```

Comment:
>str [](start = 31, length = 3)

Change the name of the variable since it's not always a string anymore. Maybe `suppression`